airflow/configuration.py

```diff
@@ -136,30 +139,121 @@ def _default_config_file_path(file_name: str) -> str:
     return os.path.join(templates_dir, file_name)
 
 
-def default_config_yaml() -> dict[str, Any]:
+def default_configuration_description(
+    include_airflow: bool = True,
+    include_providers: bool = True,
+    selected_provider: str | None = None,
+    config_file_name: str = "config.yml",
+) -> dict[str, dict[str, Any]]:
     """
-    Read Airflow configs from YAML file.
+    Read Airflow configs from YAML file, and merge them with provider configs optionally.
 
+    :param include_airflow: Include Airflow configs
+    :param include_providers: Include provider configs
+    :param selected_provider: If specified, include selected provider only
+    :param config_file_name: name of the file in "config_templates" directory to read default config from
     :return: Python dictionary containing configs & their info
     """
-    with open(_default_config_file_path("config.yml")) as config_file:
-        return yaml.safe_load(config_file)
+    base_configuration_description: dict[str, dict[str, Any]] = {}
+    if include_airflow:
+        with open(_default_config_file_path(config_file_name)) as config_file:
+            base_configuration_description.update(yaml.safe_load(config_file))
+    if include_providers:
+        from airflow.providers_manager import ProvidersManager
+
+        for provider, config in ProvidersManager().provider_configs:
+            if selected_provider and provider != selected_provider:
+                continue
+            base_configuration_description.update(config)
+    return base_configuration_description
+
+
+def dejinjafy(value: Any) -> str:
+    if isinstance(value, str):
+        if "{{{{" in value:
+            return value.replace("{{{{", "{{").replace("}}}}", "}}")
+        return value.replace("{{", "{").replace("}}", "}")
+    return value
```

Comment:
We are just concerned with up to 4 curlies right? This is simpler:

```suggestion
    if not isinstance(value, str):
        return value
    return value.replace("{{", "{").replace("}}", "}")
```

This does the 4->2 and 2->1.

Reply:
I looked at that line for so long, convinced there was a better way LOL

Reply:
I do it WAY better in the final version of those. Wait for it :).